romancal/tweakreg/tweakreg_step.py

```diff
@@ -167,166 +151,174 @@ def process(self, input):
+                y = catalog["y"]
+                if bb is None:
+                    r, d = image_model.meta.wcs(x, y)
+                    mask = np.isfinite(r) & np.isfinite(d)
+                    catalog = catalog[mask]
+
+                    n_removed_src = np.sum(np.logical_not(mask))
+                    if n_removed_src:
+                        self.log.info(
+                            f"Removed {n_removed_src} sources from {filename}'s "
+                            "catalog whose image coordinates could not be "
+                            "converted to world coordinates."
+                        )
+                else:
+                    # assume image coordinates of all sources within a bounding box
+                    # can be converted to world coordinates.
+                    ((xmin, xmax), (ymin, ymax)) = bb
+                    mask = (x > xmin) & (x < xmax) & (y > ymin) & (y < ymax)
+                    catalog = catalog[mask]
+
+                    n_removed_src = np.sum(np.logical_not(mask))
+                    if n_removed_src:
+                        self.log.info(
+                            f"Removed {n_removed_src} sources from {filename}'s "
+                            "catalog that were outside of the bounding box."
+                        )
 
-            # set meta.tweakreg_catalog
-            image_model.meta["tweakreg_catalog"] = catalog.as_array()
+                # set meta.tweakreg_catalog
+                image_model.meta["tweakreg_catalog"] = catalog.as_array()
 
-            nsources = len(catalog)
-            if nsources == 0:
-                self.log.warning(f"No sources found in {filename}.")
-            else:
-                self.log.info(f"Detected {len(catalog)} sources in {filename}.")
+                nsources = len(catalog)
+                if nsources == 0:
+                    self.log.warning(f"No sources found in {filename}.")
+                else:
+                    self.log.info(f"Detected {len(catalog)} sources in {filename}.")
 
-            images[i] = image_model
+                images[i] = image_model
 
         # group images by their "group id":
-        grp_img = list(images.models_grouped)
+        group_indices = images.group_indices
 
         self.log.info("")
-        self.log.info(f"Number of image groups to be aligned: {len(grp_img):d}.")
+        self.log.info(f"Number of image groups to be aligned: {len(group_indices):d}.")
         self.log.info("Image groups:")
 
-        if len(grp_img) == 1 and not ALIGN_TO_ABS_REFCAT:
-            self.log.info("* Images in GROUP 1:")
-            for im in grp_img[0]:
-                self.log.info(f"     {im.meta.filename}")
-            self.log.info("")
+        if len(group_indices) == 1 and not ALIGN_TO_ABS_REFCAT:
+            # self.log.info("* Images in GROUP 1:")
+            # for im in grp_img[0]:
+            #     self.log.info(f"     {im.meta.filename}")
+            # self.log.info("")
```

Comment:
I presume that this and related logging issues are a temporary expedient rather than a proposed change?

Reply:
This is a partially proposed change at the moment. It does highlight one pattern that has some negative consequences when the `ModelLibrary` is `on_disk`. The logging commented out here (if the library is `on_disk`) will result in every model being read from disk to print out the `meta.filename` of each model. This is shortly followed by every model then being re-opened to set the `cal_step` status. I think a more complete change would be to combine these loops or remove the logging if printing out the filenames isn't that important (I don't think most steps do this).